prusti-server/src/client.rs

```diff
@@ -4,58 +4,76 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-use crate::VerificationRequest;
+use crate::{ServerMessage, VerificationRequest};
+use futures_util::{
+    sink::SinkExt,
+    stream::{Stream, StreamExt},
+};
 use prusti_common::config;
-use reqwest::Client;
-use url::{ParseError, Url};
-use viper::VerificationResult;
+use tokio_tungstenite::{
+    connect_async,
+    tungstenite::{error::Error, Message},
+};
+use url::Url;
 
-pub struct PrustiClient {
-    client: Client,
-    server_url: Url,
-}
+pub struct PrustiClient;
 
 impl PrustiClient {
-    pub fn new<S: ToString>(server_address: S) -> Result<Self, ParseError> {
+    pub async fn verify<S: ToString>(
+        server_address: S,
+        request: VerificationRequest,
+    ) -> impl Stream<Item = ServerMessage> {
+        // TODO: do proper error handling
```

Comment:
How "improper" is it? For example, is it just resilience against the server unexpectedly quitting? If we panic in such a case that might be ok.

Reply:
Atm, we panic if the server address is invalid and if the connection is torn down/fails somehow unexpectedly. However, this was basically the same before: just that there this method returned a result which was `.expect`ed.